packages/components/src/components/permissions/PermissionsRole.tsx

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright (c) 2018-2019 LabKey Corporation. All rights reserved. No portion of this work may be reproduced in
+ * any form or by any electronic or mechanical means without written permission from LabKey Corporation.
+ */
+import * as React from 'react';
+import { Col, Row } from "react-bootstrap";
+import { List } from 'immutable'
+import { Principal, SecurityAssignment, SecurityRole } from "./models";
+import { RemovableButton } from "./RemovableButton";
+import { AddRoleAssignmentInput } from "./AddRoleAssignmentInput";
+import { ExpandableContainer } from "../ExpandableContainer";
+import { naturalSort } from "../../util/utils";
+
+interface Props {
+    role: SecurityRole
+    assignments: List<SecurityAssignment>
+    typesToShow: List<string>
+    principals: List<Principal>
+    onAddAssignment: (principal: Principal, role: SecurityRole) => any
+    onRemoveAssignment: (userId: number, role: SecurityRole) => any
+    onClickAssignment: (userId: number) => any
+    selected: Principal
+}
+
+export class PermissionsRole extends React.PureComponent<Props, any> {
+
+    generateClause() {
+        return (
+            <div className="container-expandable-heading--clause">
+                <span className={'permissions-title'}>{this.props.role.displayName}</span>
+            </div>
+        )
+    }
+
+    generateLinks() {
+        const { assignments, typesToShow } = this.props;
+        let count = 0;
+        if (assignments && assignments.size > 0) {
+            count = typesToShow.size === 1
+                ? assignments.filter((assignment) => SecurityAssignment.isTypeMatch(assignment.type, typesToShow.get(0))).size
+                : assignments.size;
```

Comment:
if `typesToShow.size` is not 1, is it guaranteed to be the same as the size of `assignments`?  Why is the filter not required when there's more than one type to show?

Reply:
This was just me being lazy I guess since there are only two types: 'u' and 'g'. I'll clean up this code.